src/collections.ts

```diff
@@ -0,0 +1,147 @@
+import * as collections from "@opencreek/deno-std-collections"
+import { apply, extendProtoype } from "./extendPrototype"
+
+type Collections = typeof collections
+
+function filterNotNullish<T>(this: Array<T>): Array<NonNullable<T>> {
+  return this.filter((it): it is NonNullable<T> => it != undefined)
+}
+
+function extendArray<
+  KEY extends {
+    [P in keyof Collections]: Collections[P] extends (
+      first: any[],
+      ...args: any
+    ) => any
+      ? P
+      : never
```

Comment:
Can you check if this also works when replacing `any` with `unknown`? `any` disables typechecking and matches shapes that you would not expect it to match sometimes.

Reply:
it does not work